### DISCUSSION TITLE
PrimeFaces.ajax.Request.handle does not cause validation of TabView anymore

### DISCUSSION THREAD

**Discussion by mkomko** [primefaces/community#123] [PrimeFaces] [answered]
Hi!

We are currently migrating from PrimeFaces 7 to PrimeFaces 13 and previously used the following JavaScript to manually trigger validation when switching the tabs of a _TabView_ component. Sadly, it's not working anymore, and I don't know why. I would appreciate the help of anyone with the relevant PrimeFaces JS knowledge 😀 

Reproducer: https://github.com/mkomko/primefaces-test/tree/tabview-ajax-handle-validationfailed

Validation should be triggered when switching tabs and an alert should be displayed, but it never happens.

JavaScript:
```javascript
$(function () {
            $.extend(PF("widget"), {
                onTabChange: function (tabIndex) {
                    let $this = this;

                    PrimeFaces.ajax.Request.handle({
                        source: $this.id,
                        process: $this.panelContainer.children().eq($this.cfg.selected).attr("id"),
                        update: $this.id,
                        oncomplete: function (xhr, status, args) {
                            if (args.validationFailed) {
                                alert("Validation failed!")
                            }
                        }
                    });
                }
            });
        });
```

XHTML:
```xml
<p:tabView widgetVar="widget"
                   onTabChange="return PF('widget').onTabChange(index)">

            <p:ajax event="tabChange"/>

            <p:tab title="1">
                <p:outputLabel value="1" for="text1"/>
                <p:inputText id="text1" value="#{testView.text1}" required="true"/>
                <p:message for="text1">
                    <p:autoUpdate/>
                </p:message>
            </p:tab>
            <p:tab title="2">
                <p:outputLabel value="2" for="text2"/>
                <p:inputText id="text2" value="#{testView.text2}" required="true"/>
                <p:message for="text2">
                    <p:autoUpdate/>
                </p:message>
            </p:tab>
            <p:tab title="3">
                <p:outputLabel value="3" for="text3"/>
                <p:inputText id="text3" value="#{testView.text3}" required="true"/>
                <p:message for="text3">
                    <p:autoUpdate/>
                </p:message>
            </p:tab>
        </p:tabView>
```

Thank you very much in advance!

What we are trying to achieve in the end and had working using PrimeFaces 7:
* Have a _TabView_ with dynamic tabs and cache false
* Validate when switching tabs and prevent switching tabs if validation fails
* I have seen the new attributes _focusOnLastActiveTab_ und _focusOnError_ and have dabbled with them, but without much success, especially with dynamic tabs (maybe somehow related to https://github.com/primefaces/primefaces/issues/4485)

**Comment by FlipWarthog** [comment]
I pulled your reproducer down and it's working? Am I missing something?

![image](https://github.com/primefaces/community/assets/83613837/4ab2709e-9507-46f1-9b45-53d62a43255d)

**Comment by mkomko** [comment]
@FlipWarthog Thank you very much for your answer. After testing again I found out that it works using Mojarra, but not using MyFaces! 

Pinging MyFaces experts @tandraschko @melloware 😁

**Reply by melloware** [reply, marked as answer]
@mkomko can you open a MyFaces bug.  I just compared the two response XML and MyFaces is stripping out some data.

I shortened it for clarity

**Mojarra:**
```xml
<partial-response>
	<changes>
		<update id="frmTest:j_idt7"></div><script id="frmTest:j_idt7_s" type="text/javascript">PrimeFaces.cw("TabView","widget",{id:"frmTest:j_idt7",onTabChange:function(index){return PF('widget').onTabChange(index)},effectDuration:"normal",scrollable:false});</script>]]></update>
		<update id="j_id1:jakarta.faces.ViewState:0"><![CDATA[5842890641509202258:5325669557210236122]]></update>
		<extension ln="primefaces" type="args">{&#34;validationFailed&#34;:true}</extension>
	</changes>
</partial-response>
```

**MyFaces:**
```xml
<partial-response id="j_id__v_0">
	<changes>
		<update id="frmTest:j_id_9"><script id="frmTest:j_id_9_s" type="text/javascript">PrimeFaces.cw("TabView","widget",{id:"frmTest:j_id_9",onTabChange:function(index){return PF('widget').onTabChange(index)},effectDuration:"normal",scrollable:false});</script>]]></update>
		<update id="j_id__v_0:jakarta.faces.ViewState:1"><![CDATA[Y2NiMjlmN2E2YjhjMjg3MDAwMDAwMDAx]]></update>
	</changes>
</partial-response>
```

The important part that is missing in MyFaces is this..

```xml
<extension ln="primefaces" type="args">{&#34;validationFailed&#34;:true}</extension>
```

It is similar to this but different: https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4536

**Reply by mkomko** [reply]
I'll do that, thank you very much!

**Reply by mkomko** [reply]
I opened a Jira issue here: https://issues.apache.org/jira/browse/MYFACES-4629

**Comment by tandraschko** [comment]
@melloware are you sure validationFailed is called on server side?

**Reply by melloware** [reply]
Actually good point I didn't dig that far into it once I realized it wasn't coming back in the partial response.

**Comment by melloware** [comment]
[pf-tabview.zip](https://github.com/primefaces/community/files/12749424/pf-tabview.zip)
Reproducer:

**Reply by melloware** [reply]
@mkomko sorry didn't get to this today did you debug to see if the validation is failing in the backend but just getting stripped by MyFaces or it's not getting there at all?

**Reply by mkomko** [reply]
@melloware I did not and I wouldn't know where to start 😐
